[DO NOT MERGE] Documentation update review (#172)
canonical/kafka-operator: docs/how-to/h-manage-units.md

# How to manage units

> Apache Kafka ships with `bin/*.sh` commands to do various administrative tasks, among which

Reviewer: ```suggestion
Apache Kafka ships with `bin/*.sh` commands to do various administrative tasks such as:
```

* `bin/kafka-config.sh` to update cluster configuration
* `bin/kafka-topics.sh` for topic management
* `bin/kafka-acls.sh` for management of ACLs of Kafka users

Please refer to the upstream [Kafka project](https://github.com/apache/kafka/tree/trunk/bin), 
for a full list of the bash commands available in Kafka distributions. Also, you can 
use `--help` argument for printing a short summary of the argument for a given 
bash command. 

The most important commands are also exposed via the [Charmed Kafka snap](https://snapcraft.io/charmed-kafka), 

> accessible via `charmed-kafka.<command>`. Please refer to [this table](TODO) for 

Author: as before.

Commands can also be run within a Kafka broker, since both the authentication 
file (alongside with truststore if needed) and the Charmed Kafka snap are 
already present. 

#### Example (Listing topics)

For instance, in order to list the current topics on the Kafka cluster, you can run:
```
juju ssh kafka/leader 'charmed-kafka.topics --bootstrap-server $BOOTSTRAP_SERVERS --list --command-config /var/snap/charmed-kafka/common/client.properties'
```

### Juju External users

For external users managed by the  [Data Integrator Charm](https://charmhub.io/data-integrator), 
the endpoints and credentials can be fetched using the dedicated action

```shell
juju run data-integrator/leader get-credentials --format yaml
```

The `client.properties` file can be generated by replacing relevant information in the 
one available on the brokers, e.g. 

```
BOOTSTRAP_SERVERS=$(juju run data-integrator/leader get-credentials --format yaml | yq .kafka.endpoints )
USERNAME=$(juju run data-integrator/leader get-credentials --format yaml | yq .kafka.username )
PASSWORD=$(juju run data-integrator/leader get-credentials --format yaml | yq .kafka.password )
```

> ## Replication and Scaling

Reviewer: I think this should be moved up so it's the first section. It's the default instructions on how to scale with juju, in case that's all the user is looking for, and then more complex management details can follow for more advanced users. 
What do you think?

Increasing the number of Kafka brokers can simply be achieved by adding more units
to the Charmed Kafka application, e.g. 

```shell
juju add-unit kafka -n <num_brokers_to_add>
```

> For more information on how to manage units, please refer to [this](https://juju.is/docs/juju/manage-units)

Reviewer: ```suggestion
For more information on how to manage units, please refer to the [Juju documentation](https://juju.is/docs/juju/manage-units)
```
